core/commands/SubprojectDeleteCommand.php

```diff
@@ -0,0 +1,116 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with MantisBT.  If not, see <http://www.gnu.org/licenses/>.
+
+require_api( 'constant_inc.php' );
+require_api( 'config_api.php' );
+require_api( 'helper_api.php' );
+require_api( 'project_api.php' );
+require_api( 'project_hierarchy_api.php' );
+
+use Mantis\Exceptions\ClientException;
+
+/**
+ * A command that deletes a project subproject.
+ */
+class SubprojectDeleteCommand extends Command {
+	/**
+	 * $p_data['query'] is expected to contain:
+	 * - project_id (integer)
+	 * - subproject_id (integer)
+	 *
+	 * @param array $p_data The command data.
+	 */
+
+	/**
+	 * @var integer
+	 */
+	private $project_id;
+
+	/**
+	 * @var integer
+	 */
+	private $subproject_id;
+
+	/**
+	 * Constructor
+	 *
+	 * @param array $p_data The command data.
+	 */
+	function __construct( array $p_data ) {
+		parent::__construct( $p_data );
+	}
+
+	/**
+	 * Validate the data.
+	 */
+	function validate() {		
+		$this->project_id = helper_parse_id( $this->query( 'project_id' ), 'project_id' );
+
+		if( !access_has_project_level( config_get( 'manage_project_threshold' ), $this->project_id ) ) {
+			throw new ClientException(
+				'Access denied to delete subprojects',
+				ERROR_ACCESS_DENIED );
+		}
+
+		if ( config_get( 'subprojects_enabled' ) == OFF ) {
+			throw new ClientException(
+				'Access denied to delete subprojects',
+				ERROR_ACCESS_DENIED );
+		}
+
+		if( !project_exists( $this->project_id )) {
+			throw new ClientException(
+				"Project \"$this->project_id\"not found",
+				ERROR_PROJECT_NOT_FOUND,
+				array( $this->project_id ) );
+		}
+
+		$this->subproject_id = helper_parse_id( $this->query( 'subproject_id' ), 'subproject_id' );
+		if( !project_exists( $this->subproject_id )) {
+			throw new ClientException(
+				"Project \"$this->subproject_id\" not found",
+				ERROR_PROJECT_NOT_FOUND,
+				array( $this->subproject_id ) );
+		}
+		
+		if( !in_array( $this->subproject_id, project_hierarchy_get_subprojects( $this->project_id, true ) ) ) {
+			throw new ClientException(
+				"Project \"$this->subproject_id\" is not a subproject of \"$this->project_id\"",
+				ERROR_PROJECT_SUBPROJECT_NOT_FOUND,
+				array( $this->subproject_id, $this->project_id ) );
+		}		
+	}
+
+	/**
+	 * Process the command.
+	 *
+	 * @returns array Command response
+	 */
+	protected function process() {
+		if( $this->project_id != helper_get_current_project() ) {
+			# in case the current project is not the same project of the bug we are
+			# viewing, override the current project. This to avoid problems with
+			# categories and handlers lists etc.
+			global $g_project_override;
+			$g_project_override = $this->project_id;
+		}
+
+		project_hierarchy_remove( $this->subproject_id, $this->project_id );
+
+		return array( 'id' => $this->subproject_id );
```

Comment:
Not sure we need to return anything here, but if we do, we should return a `project` structure and not an `id` field.